starlette/datastructures.py

```diff
@@ -238,32 +246,34 @@ def __str__(self) -> str:
         return ", ".join(repr(item) for item in self)
 
 
-class ImmutableMultiDict(typing.Mapping):
+class ImmutableMultiDict(typing.Mapping[_KeyType, _ValueType]):
+    _dict: typing.Dict[_KeyType, _ValueType]
+
     def __init__(
         self,
         *args: typing.Union[
-            "ImmutableMultiDict",
-            typing.Mapping,
-            typing.List[typing.Tuple[typing.Any, typing.Any]],
+            "ImmutableMultiDict[_KeyType, _ValueType]",
+            typing.Mapping[_KeyType, _ValueType],
+            typing.Sequence[typing.Tuple[_KeyType, _ValueType]],
```

Comment:
![image](https://user-images.githubusercontent.com/25364828/169958431-f5c6641a-f7fe-4944-85f1-4cc116ae2226.png)

Maybe we use `typing.Iterable` is better? Just like `dict`.

Reply:
yep